setup.go

```diff
@@ -118,22 +139,124 @@ func Setup(ctx context.Context, cfg *Config) (*Node, error) {
 
 	bsctx := metri.CtxScope(ctx, "rainbow")
 
-	r1, err := routingv1client.New(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
-	if err != nil {
-		return nil, err
-	}
-
-	vs := httpcontentrouter.NewContentRoutingClient(r1)
+	var pr routing.PeerRouting
+	var vs routing.ValueStore
+	var cr routing.ContentRouting
 
 	opts = append(opts, libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
-		return vs, nil
+		if cfg.RoutingV1 != "" {
+			routingClient, err := delegatedHTTPContentRouter(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
+			if err != nil {
+				return nil, err
+			}
+			pr = routingClient
+			vs = routingClient
+			cr = routingClient
+		} else {
+			// If there are no delegated routing endpoints run an accelerated Amino DHT client and send IPNI requests to cid.contact
+
+			// TODO: This datastore shouldn't end up containing anything anyway so this could potentially just be a null datastore
+			memDS, err := levelds.NewDatastore("", nil)
+			if err != nil {
+				return nil, err
+			}
+
+			var dhtHost host.Host
+			if cfg.DHTSharedHost {
+				dhtHost = h
+			} else {
+				dhtHost, err = libp2p.New(
+					libp2p.NoListenAddrs,
+					libp2p.BandwidthReporter(bwc),
+					libp2p.DefaultTransports,
+					libp2p.DefaultMuxers,
+				)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			var standardClient *dht.IpfsDHT
+			var fullRTClient *fullrt.FullRT
+
+			if cfg.DHTType == Combined || cfg.DHTType == Standard {
+				standardClient, err = dht.New(ctx, dhtHost,
+					dht.Datastore(memDS),
+					dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+					dht.Mode(dht.ModeClient),
+				)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			if cfg.DHTType == Combined || cfg.DHTType == Accelerated {
+				fullRTClient, err = fullrt.NewFullRT(dhtHost, dht.DefaultPrefix,
+					fullrt.DHTOption(
+						dht.Validator(record.NamespacedValidator{
+							"pk":   record.PublicKeyValidator{},
+							"ipns": ipns.Validator{KeyBook: h.Peerstore()},
+						}),
+						dht.Datastore(memDS),
+						dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+						dht.BucketSize(20),
```

Comment:
does the fullRT dht understand dht.ModeClient? This nodes shouldn't be storing records for other people even if they are publicly reachable.

Reply:
Pretty sure it doesn't, but that's because it is only a client (i.e. it never operates in server mode)